File: examples/proxy/rest/nodejs/proxy.mjs
// Routes browser traffic through a residential proxy and logs the resulting IP.
//
// Run: node proxy.mjs

const response = await fetch(
  'https://production-sfo.browserless.io/scrape?token=YOUR_API_TOKEN_HERE&proxy=residential&proxyCountry=us',
  {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({
      url: 'https://api.ipify.org?format=json',
      elements: [{ selector: 'body' }],
    }),
  }
);

const { data } = await response.json();
const ip = JSON.parse(data[0].results[0].text).ip;
console.log('Proxy IP:', ip);

File: examples/proxy/rest/python/proxy.py
# Routes browser traffic through a residential proxy and logs the resulting IP.
#
# Install: pip install requests
# Run:     python proxy.py

import requests
import json

response = requests.post(
    'https://production-sfo.browserless.io/scrape',
    params={
        'token': 'YOUR_API_TOKEN_HERE',
        'proxy': 'residential',
        'proxyCountry': 'us',
    },
    json={
        'url': 'https://api.ipify.org?format=json',
        'elements': [{'selector': 'body'}],
    },
)

data = response.json()['data']
ip = json.loads(data[0]['results'][0]['text'])['ip']
print('Proxy IP:', ip)
